src/media-groups.js

```diff
@@ -269,27 +269,24 @@ export const setupListeners = {
       segmentLoaders: { [type]: segmentLoader }
     } = settings;
 
-    playlistLoader.on('loadedmetadata', () => {
+    // only start to append segments to source buffers after
+    // the main segment loader has already appended
+    settings.segmentLoaders.main.one('appended', () => {
       const media = playlistLoader.media();
 
-      segmentLoader.playlist(media, requestOptions);
```

Comment:
this needs to happen before calling load on the segmentLoader... for the alt audio loader too. Particularly with DASH

Reply:
I would think `loadedplaylist` and `loadedmetadata` would have to happen before `appended` on the main segment loader? which means playlist will be set beforehand.

Reply:
hopefully 😄  you might want to double check DASH sidx to see if this is always the case

Reply:
Refactored so that `appended`, `loadedplaylist` and `loadedmetadata` all listen with `loadPlaylist`